ProcessLib/Process.cpp

```diff
@@ -264,7 +267,7 @@ void Process::preIteration(const unsigned iter, const GlobalVector &x)
 {
     // In every new iteration cached values of secondary variables are expired.
     for (auto& cached_var : _cached_secondary_variables) {
-        cached_var->expire();
+        cached_var->updateCurrentSolution(x, *_local_to_global_index_map);
```

Comment:
`CachedSecondaryVariable`s are used to produce output. Since output can be produced after every nonlinear iteration (e.g., for debugging), the update has to be done in every iteration.
I chose to do the update in the `preIteration()` method.
But probably I could remove the update from the timestep method. TODO.